### ISSUE TITLE
Download without installing

### ISSUE THREAD

**Issue by alltheseas** [zapstore/zapstore#101]
## user story

As a Zap Store user who would like to verify, I would like a method to be able to independently of ZS verify an app's SHA256 hash, so that I can gain confidence that ZS is not rugpulling me with an incorrect APK. 

### acceptance criteria

1. ZS user can download APK *without* installing
2. ZS lists the APK file location on device
3. ~~(optional)(future) user can report that they have verified the hash to the nostr network (i.e. via ZS relay)~~

### question
1. should there be a way for user to copy or paste sha-256 value displayed by ZS and/or another app to compare hash match?

### related to
#79 sha-256 not visible

**Comment by franzaps** [comment]
I don't think anyone will do 3.

But sure. We can show file paths for people to check with DeadHash or whatever

**Comment by franzaps** [comment]
I like the idea of "curators" more than "hash verifiers". In zs you'll be able to have app curators to whom you delegate this task. On top of that you of course can verify stuff like WoT, hashes matching, etc

**Comment by alltheseas** [comment]
Will you require curators to verify hash?

**Comment by franzaps** [comment]
> Will you require curators to verify hash?

This is a thankless job, there are too many releases to expect people/curators to keep checking and validating hashes for. If users don't trust zap.store they can build it themselves from source, or not use it.

**Comment by alltheseas** [comment]
> there are too many releases to expect people/curators to keep checking and validating hashes for. 

Ok, noted, curators are not required to verify hash. 

If a curator does decide to verify hash, should this be displayed?

**Comment by franzaps** [comment]
Dude I'm not against it! And I may be wrong. It just feels like a lot of work vs the alternative. Yes we could have "hash verified" NIP or something that gets signed every time someone (curator or within web of trust) checked the hash, and display it before installation. Reminder that this is important only upon first install.

**Comment by franzaps** [comment]
Hold on, I got confused with the latest comment about other users verifying hashes. I think the original user story is a good one. I already implemented #79 so it would be a matter of exposing the downloaded file. With those two pieces user can manually check with DeadHash for example

**Comment by keith-gardner** [comment]
1 and 2 are highly useful.

Branta allows users to drag & drop PGP verify zeus .apks (more to follow, this is only partially available). We can also sweep folders.

My ideal flow would be:

1. I find Zeus .apk on ZS
2. ZS downloads .apk to /some/default/path
3. Branta is sweeping that path for .apks from zeus
4. Branta PGP verifies each file in /some/default/path on `30s` cadence
5. I manually shuttle the .apk to my android.



This way:
- No central app stores, ZS facilitates discovery
- Branta tells me the .apk is PGP signed, in addition to WOT recommendation
- I can send this file to all of my androids, other friends, etc.


I think as ZS bootstraps, its nice to have barebones PGP underlying in the process.

**Comment by franzaps** [comment]
Agree, PGP verification is useful especially during this bootstrapping phase.

Trust signals on zap.store are multiple (currently WoT and namespaces - i.e. github.com/user/repo). PGP linking will be coming soon via https://github.com/nostr-protocol/nips/pull/1182 , requires work on dev tooling coming after 0.2.

My ideal UX is having all verification/installation without leaving zap.store.

I checked out branta which I think will be perfect for a validator we're planning with DVMs, see https://github.com/zapstore/zapstore/issues/23 . As a centralized provider you attest to the authenticity of a certain package, can charge for it all based on your reputation on a specific niche (bitcoin wallets), which in turn could be powered by webs of trust. For example: Alice and Bob trust Branta, pay them a few sats to verify authenticity of this wallet (on top of other trust signals present in zap.store)

**Comment by franzaps** [comment]
Won't implement the download unless more users request it. For now, it's possible to copy the APK URL and the hash and verify with external tools if needed
